### PR TITLE
feat(databricks): init terraform databricks rules

### PR DESCRIPTION
**Pull request by dim-ops**
### Context
Init Databricks Terraform Providers with some rules.
I mainly followed Databricks Best Pratices => https://docs.databricks.com/clusters/cluster-config-best-practices.html and worked with Data Engineer to validate my rules.

### Proposed Changes
Add a new provider and new rules

### Roadmap
1. Write rules and tests ✅
2. Review by Data Engineer ✅ (Did it once)
3. Review by my teams ✅
4. Review by Databricks 🔜 (Bedrock is partner we will ask them ; The first mail was sent, we had a response from Databricks they are excited about the subject, we will make a check soon.) => It takes too long to get an appointment...
5. Review by cx

### Rules

image

### How to test
I created some tests inside each folder.

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Review by cxMiguelSilva — COMMENTED** [review]
Hi @dim-ops, hope you are doing great!
After the question, you have added to our discussions we have decided that databricks should not be deemed as a cloud provider since it makes use of was, gpc and azure.
So the queries should be set to common on the attribute cloud provider on the metadata file.
Sorry for the late response, and inconvenience.
Let us know if you have any other questions.

**Comment by dim-ops** [comment]
Hi @cxMiguelSilva I had a doubt about this too but it's noted, thank you :)

File: assets/queries/terraform/databricks/cluster_aws_attributes/query.rego [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package Cx
+
+import data.generic.terraform as tf_lib
+
+#use spot instance
+CxPolicy[result] {
```
Comment (dim-ops) [inline assets/queries/terraform/databricks/cluster_aws_attributes/query.rego:6]: SPOT

File: assets/queries/terraform/databricks/autoscale_not_setup/query.rego [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package Cx
+
+import data.generic.terraform as tf_lib
+
+CxPolicy[result] {
+	resource := input.document[i].resource.databricks_cluster[name]
+	not resource.autoscale
```
Comment (dim-ops) [inline assets/queries/terraform/databricks/autoscale_not_setup/query.rego:7]: To remove

File: assets/queries/terraform/databricks/cluster_azure_attributes/test/positive_expected_result.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[
+	{
```
[the remark this answers is not in the document]
Comment (dim-ops) [reply assets/queries/terraform/databricks/cluster_azure_attributes/test/positive_expected_result.json:2]: Like in every `positive_expected_result.json` file

File: assets/queries/terraform/databricks/not_use_lts_spark_version/query.rego [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("'databricks_spark_version[%s]' should be a LTS version'", [name]),
+		"keyActualValue": sprintf("'databricks_spark_version[%s]' is not a LTS version'", [name]),
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i].resource.databricks_cluster[name]
+
+	not contains(resource.spark_version, "LTS"); not contains(resource.spark_version, "data")
```
[the remark this answers is not in the document]
Comment (dim-ops) [reply assets/queries/terraform/databricks/not_use_lts_spark_version/query.rego:24]: As seen together ";" corresponds to an and

I improved this rules by `not contains(resource.spark_version, "data.databricks_spark_version")`

**Comment by dim-ops** [comment]
Hello @cxMiguelSilva 👋 

It's a bit long to have a review of Databricks, is it possible that you give me some feedback on my PR even if it's in draft?

I don't have any other rules to add for the moment :)

**Comment by gabriel-cx** [comment]
Hi @dim-ops,
Thank you for your contribution here! 😃
Sure, we will take a look and we will give you feedback ASAP!

**Comment by dim-ops** [comment]
> Thanks for the PR, the code looks good. We also want feedback from our AppSec team as part of the review, so some patience is needed.
> 
> In the meanwhile, can you rebase your branch, so it would be ready for merge?

It's done :)

**Comment by dim-ops** [comment]
Of course, it needs to fix conflicts, but do you when this PR can be merged ? @kaplanlior

**Comment by gabriel-cx** [comment]
Hi @dim-ops ,

Since AppSec teams resources are focused to tackle internal projects, we couldn't review all your queries yet.
However, we are working on this **[new feature called "Experimental Features"](https://github.com/Checkmarx/kics/pull/6614)**. This will allow, by using a flag, to run queries that are not fully reviewed yet (like yours regarding Databricks)! ^^ 

In this case, when we finish this feature, we can merge your **amazing** contribution and '_**hide**_' it behind this new flag. 

The goal of this feature is, for example:
If a user **uses the flag '_experimental-features_'**, your **Databricks queries will be used by the KICS scan**, if the **flag is not used**, **Databricks queries will stay** '_**inactive**_'.
When your queries are fully reviewed, we will update this flag behavior in order to make your queries no longer '**_inactive_**' behind the flag, so they will be used by KICS by default, like any other queries.

P.S: Feel free to contribute in this **[new feature](https://github.com/Checkmarx/kics/pull/6614)** as well if you interested! :D 

**Thank you again for this contribution!**

**Comment by dim-ops** [comment]
Thank you @gabriel-cx, I'll take the time to help you if I can :)

**Comment by gabriel-cx** [comment]
Hi @dim-ops ,

We plan to merge this [PR ](https://github.com/Checkmarx/kics/pull/6614) and make it available in the next KICS version (1.7.9).
It means, in the next weeks we will try to review your PR and merge it and make it available in the KICS version 1.7.10! :D
In the next 2 weeks we will only ask you to perform a little change so your feature can be merged. We will keep you updated. Thanks again for the help!

**Comment by gabriel-cx** [comment]
Hi @dim-ops ,

Hope this message finds you well!

I'm happy to announce that soon we will be able to merge your PR! :D
We want to ask you to add the databricks logo under the "Beta Features" [here ](https://github.com/Checkmarx/kics/blob/master/README.md?plain=1#L60) and [here](https://github.com/Checkmarx/kics/blob/master/docs/index.md?plain=1#L85). You can just remove the "Soon..." and add your logo.

Thank you again!!

**Comment by gabriel-cx** [comment]
Hi  @dim-ops ,

Thanks for the update!

Kindly add 'databricks' in the same files as this other [PR](https://github.com/Checkmarx/kics/pull/6314/files). With that we make sure the Engine will 'fully recognize' databricks. :)

**Comment by gabriel-cx** [comment]
Hi @dim-ops ,

Thanks for the update!
Kindly add the databricks queries path[ in this test file](https://github.com/Checkmarx/kics/blob/master/test/main_test.go#L71).

**Comment by gabriel-cx** [comment]
@dim-ops , looks like your unit-tests are failing due to this:

image

**Comment by dim-ops** [comment]
@gabriel-cx I fixed the duplicate ID. I see some tests failed, but I think it's not related about my PR, no ?

**Comment by gabriel-cx** [comment]
Hi @dim-ops ,

I will check what can be the root cause.

Currently, we changed the way we want to approach the "Beta features". This means, we are waiting this [PR](https://github.com/Checkmarx/kics/pull/6769) to be merged so we can merge yours as well. I will keep you updated!

**Comment by gabriel-cx** [comment]
Hi @dim-ops ,

Looks like your unit tests are failing due to this 3 highlighted tests:
image

**Comment by dim-ops** [comment]
@gabriel-cx everything seems good :)

**Comment by gabriel-cx** [comment]
@dim-ops looks cool, thanks! :D 
Today we will release a new KICS version (v1.7.11) and your changes can be part of the next release (v1.7.12).

**The only changes we'd like to ask you is, kindly add `"experimental": "true"` in every _metadata.json_ files of every query.**

KICS will have a new flag `--experimental-queries`, this means, KICS will only run queries that have the `"experimental": "true"` if the user runs KICS with that flag.
Since our AppSec team didn't review yet your queries, that's why we want to set them as "experimental". When our AppSec team review your queries and agree with them, we will remove the `"experimental": "true"` field from the metadata and the queries will became official and not experimental. ^^

**Review by gabriel-cx — APPROVED** [review]

**Comment by gabriel-cx** [comment]
Thank you @dim-ops ! It's merged! :D

**Comment by gabriel-cx** [comment]
Hi @dim-ops ,

I'm happy to let you know that your databricks queries are now part of the KICS 'official queries'!
This means, databricks queries will run by default when KICS detects terraform files. 
This functionality will start to be available in the next KICS version (v2.1.0).

Thank you again for all your help on this!

**Pull request merged**